### ISSUE TITLE
Tests?

### ISSUE THREAD

**Issue by rwjblue** [embroider-build/embroider#1]
Not sure if I've missed something in the layout, but I'm curious how you've been testing as there don't seem to be tests in the repo at all...

**Comment by ef4** [comment]
Yes, there are no automated tests here yet. The reason is that my standard of correctness is whatever the hell existing addons and apps do, and I'm still in the process of characterizing that. I deliberately don't want to be one of those projects that has a lovely test suite but then an extremely painful integration into real apps. So I have overweighted on testing in real apps.

We will absolutely want to crystalize those semantics in the form of a test suite, and in principle it would be great to do little by little as we go, but deciding what "correct" looks like has still been undergoing pretty big churn as I find more "fun" parts of the existing ecosystem.

Right now this repo is an impenetrable wall of code with literally nothing that is easy to run. That is why I dragged my feet on sharing it -- I know it's not usable by anybody else yet. But people were clamoring to see it.
